x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,331 @@
+
+      const {
+        singleMetricViewerControlsApi,
+        serializeSingleMetricViewerState,
+        singleMetricViewerComparators,
+        onSingleMetricViewerDestroy,
+      } = initializeSingleMetricViewerControls(state, titlesApi);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+
+      const refresh$ = new BehaviorSubject<void>(undefined);
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...timeRangeApi,
+          ...singleMetricViewerControlsApi,
+          dataLoading,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeSingleMetricViewerState(),
+              },
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...singleMetricViewerComparators,
+        }
+      );
+
+      const { singleMetricViewerData$, onDestroy } = initializeSingleMetricViewerDataFetcher(
+        api,
+        dataLoading,
+        blockingError,
+        refresh$,
+        services[1].data.query.timefilter.timefilter
+      );
+
+      return {
+        api,
+        Component: () => {
+          const [chartWidth, setChartWidth] = useState<number>(0);
+          const [zoom, setZoom] = useState<AppStateZoom | undefined>();
+          const [selectedForecastId, setSelectedForecastId] = useState<string | undefined>();
+          const [selectedJob, setSelectedJob] = useState<MlJob | undefined>();
+          const [jobsLoaded, setJobsLoaded] = useState(false);
+          const [error, setError] = useState<string | undefined>();
+
+          const {
+            mlApiServices,
+            mlJobService,
+            mlTimeSeriesExplorerService,
+            toastNotificationService,
+          } = services[2];
+          const I18nContext = services[0].i18n.Context;
+          const theme = services[0].theme;
+          const datePickerDeps: DatePickerDependencies = {
+            ...pick(services[0], ['http', 'notifications', 'theme', 'uiSettings', 'i18n']),
+            data: services[1].data,
+            uiSettingsKeys: UI_SETTINGS,
+            showFrozenDataTierChoice: false,
+          };
+
+          const { singleMetricViewerData, bounds, lastRefresh } =
+            useStateFromPublishingSubject(singleMetricViewerData$);
+
+          useUnmount(() => {
+            onSingleMetricViewerDestroy();
+            onDestroy();
+            subscriptions.unsubscribe();
+          });
+
+          useReactEmbeddableExecutionContext(
+            services[0].executionContext,
+            // TODO https://github.com/elastic/kibana/issues/180055
+            // @ts-ignore
+            parentApi?.executionContext?.value ?? { name: 'dashboard' },
```

Comment:
the issue has been resolved, could you please remove this temp workaround? you can look up at the anomaly swim lane

Reply:
Removed in 04cb22e2814cf01abb6fa214c58acd8f459f4034